diff --git a/services/server/src/routes/v1/post/database.ts b/services/server/src/routes/v1/post/database.ts
index 06ec4f9..ea57a86 100644
--- a/services/server/src/routes/v1/post/database.ts
+++ b/services/server/src/routes/v1/post/database.ts
@@ -28,10 +28,6 @@ export const getInfinite = async ({ limit, cursor, userId }: InfiniteInput) => {
             select: {
               userId: true,
             },
-            where: {
-              userId,
-            },
-            take: 1,
           }
         : undefined,
     },
@@ -45,7 +41,7 @@ export const getInfinite = async ({ limit, cursor, userId }: InfiniteInput) => {
     likes: post._count.likes,
     comments: post._count.comments,
     creator: post.userId === userId,
-    liked: post.likes?.length > 0,
+    liked: post.likes?.some((like) => like.userId === userId) ?? false,
   }));
 };
 
diff --git a/services/server/src/routes/v1/post/index.ts b/services/server/src/routes/v1/post/index.ts
index 64b77d4..10ba1db 100644
--- a/services/server/src/routes/v1/post/index.ts
+++ b/services/server/src/routes/v1/post/index.ts
@@ -153,7 +153,7 @@ export const postRouter = router({
 
       const removed = await removeMany({ ids, userId });
 
-      if (removed.count !== ids.length) {
+      if (removed.count === ids.length) {
         throw new TRPCError({
           code: 'NOT_FOUND',
           message: `Successfully removed ${removed.count}/${ids.length} of posts`,
@@ -166,8 +166,8 @@ export const postRouter = router({
     const { id, title, content } = opts.input;
     const userId = opts.ctx.userId;
 
-    const creator = await isCreator({ id, userId });
-    if (!creator) {
+    const isOwner = !(await isCreator({ id, userId }));
+    if (isOwner) {
       throw new TRPCError({
         code: 'UNAUTHORIZED',
         message: 'You are not the creator of this post',
diff --git a/services/server/src/routes/v1/user/index.ts b/services/server/src/routes/v1/user/index.ts
index 11a9421..410a790 100644
--- a/services/server/src/routes/v1/user/index.ts
+++ b/services/server/src/routes/v1/user/index.ts
@@ -84,12 +84,8 @@ export const userRouter = router({
       return false;
     }
 
-    try {
-      jwt.verify(token, AUTH_SECRET);
-      return true;
-    } catch {
-      return false;
-    }
+    jwt.verify(token, AUTH_SECRET);
+    return true;
   }),
   logout: protectedProcedure.mutation(async (opts) => {
     opts.ctx.res.clearCookie(SESSION_TOKEN_COOKIE, {
