diff --git a/services/server/src/index.ts b/services/server/src/index.ts
index b5a2278..e420093 100644
--- a/services/server/src/index.ts
+++ b/services/server/src/index.ts
@@ -11,7 +11,6 @@ import morgan from 'morgan';
 import path from 'path';
 
 import {
-  CLIENT_URL,
   COOKIE_SECRET,
   CSRF_SECRET,
   PORT,
@@ -40,7 +39,7 @@ async function main() {
 
   app.use(
     cors({
-      origin: CLIENT_URL,
+      origin: true,
       credentials: true,
     })
   );
diff --git a/services/server/src/routes/v1/post/comment/index.ts b/services/server/src/routes/v1/post/comment/index.ts
index cfaae1e..d7d034b 100644
--- a/services/server/src/routes/v1/post/comment/index.ts
+++ b/services/server/src/routes/v1/post/comment/index.ts
@@ -9,6 +9,7 @@ import {
   removeManyInput,
 } from '@shared/zod/comment.js';
 
+import { prisma } from '@sc/lib/prisma.js';
 import { StatusCode } from '@sc/utils/status-code.js';
 
 import {
@@ -145,6 +146,27 @@ export const commentRouter = router({
     await remove({ id });
     opts.ctx.res.status(StatusCode.OK);
   }),
+  removeOwn: protectedProcedure.input(removeInput).mutation(async (opts) => {
+    const { id } = opts.input;
+    const userId = opts.ctx.userId;
+
+    const comment = await prisma.comment.findFirst({
+      where: {
+        id,
+        userId,
+      },
+    });
+
+    if (!comment) {
+      throw new TRPCError({
+        code: 'UNAUTHORIZED',
+        message: 'You are not the creator of this comment',
+      });
+    }
+
+    await remove({ id });
+    opts.ctx.res.status(StatusCode.OK);
+  }),
   removeMany: protectedProcedure
     .input(removeManyInput)
     .mutation(async (opts) => {
diff --git a/services/server/src/routes/v1/post/database.ts b/services/server/src/routes/v1/post/database.ts
index 06ec4f9..5e1c795 100644
--- a/services/server/src/routes/v1/post/database.ts
+++ b/services/server/src/routes/v1/post/database.ts
@@ -67,6 +67,14 @@ export const getInfiniteCreator = async ({
       _count: {
         select: { likes: true, comments: true },
       },
+      comments: {
+        select: {
+          id: true,
+          likes: {
+            select: { id: true },
+          },
+        },
+      },
     },
     where: {
       userId,
diff --git a/services/server/src/routes/v1/post/index.ts b/services/server/src/routes/v1/post/index.ts
index 64b77d4..640f207 100644
--- a/services/server/src/routes/v1/post/index.ts
+++ b/services/server/src/routes/v1/post/index.ts
@@ -113,14 +113,6 @@ export const postRouter = router({
     const { id } = opts.input;
     const userId = opts.ctx.userId;
 
-    const liked = await hasLiked({ id, userId });
-    if (!liked) {
-      throw new TRPCError({
-        code: 'CONFLICT',
-        message: 'You have not liked this post',
-      });
-    }
-
     await unlike({ id, userId });
     opts.ctx.res.status(StatusCode.OK);
   }),
