diff --git a/services/client/src/core/components/Providers/SessionProvider.tsx b/services/client/src/core/components/Providers/SessionProvider.tsx
index 54a2aff..dc73c47 100644
--- a/services/client/src/core/components/Providers/SessionProvider.tsx
+++ b/services/client/src/core/components/Providers/SessionProvider.tsx
@@ -29,7 +29,7 @@ export default function SessionProvider({
   };
 
   useEffect(() => {
-    if (!isPending && !user) {
+    if (!user) {
       const localUser = getItem(StorageType.USER);
       if (localUser) {
         if (!data) {
diff --git a/services/client/src/core/utils/bigint.ts b/services/client/src/core/utils/bigint.ts
index 63f14a1..45deafe 100644
--- a/services/client/src/core/utils/bigint.ts
+++ b/services/client/src/core/utils/bigint.ts
@@ -9,4 +9,4 @@ export const parseBigInt = (value: UnparsedDefaultId) => defaultId.parse(value);
 export const compareEqual = (a: UnparsedDefaultId, b: UnparsedDefaultId) =>
   parseBigInt(a) === parseBigInt(b);
 export const compareLessThan = (a: UnparsedDefaultId, b: UnparsedDefaultId) =>
-  parseBigInt(a) < parseBigInt(b);
+  parseBigInt(a) > parseBigInt(b);
diff --git a/services/client/src/pages/_components/Post/List.tsx b/services/client/src/pages/_components/Post/List.tsx
index 3982a17..94846a9 100644
--- a/services/client/src/pages/_components/Post/List.tsx
+++ b/services/client/src/pages/_components/Post/List.tsx
@@ -1,4 +1,4 @@
-import { lazy, useMemo } from 'react';
+import { lazy } from 'react';
 
 import InfiniteScrollWindow from '@cc/components/InfiniteScrollWindow';
 import { t } from '@cc/lib/trpc';
@@ -13,10 +13,7 @@ export default function List() {
     }
   );
   const { data } = infinitePosts;
-  const allRows = useMemo(
-    () => (data ? data.pages.flatMap((d) => d.posts) : []),
-    [data]
-  );
+  const allRows = data ? data.pages.flatMap((d) => d.posts) : [];
 
   return (
     <InfiniteScrollWindow
diff --git a/services/server/src/routes/v1/trpc/middleware.ts b/services/server/src/routes/v1/trpc/middleware.ts
index 43f710d..5daa2e4 100644
--- a/services/server/src/routes/v1/trpc/middleware.ts
+++ b/services/server/src/routes/v1/trpc/middleware.ts
@@ -11,9 +11,6 @@ export const trpcMiddleware = createExpressMiddleware({
   onError({ error }) {
     if (error.code === 'INTERNAL_SERVER_ERROR') {
       lerror('Internal server error:', error);
-
-      // Incase of internal server error, we don't want to leak the error to the client
-      error.message = 'Internal server error';
     }
   },
 });
